core/src/style/pointStyleBuilder.cpp

```diff
@@ -138,11 +138,17 @@ auto PointStyleBuilder::applyRule(const DrawRule& _rule, const Properties& _prop
     _rule.get(StyleParamKey::angle, p.labelOptions.angle);
 
     auto sizeParam = _rule.findParameter(StyleParamKey::size);
-    if (sizeParam.stops && sizeParam.value.is<float>()) {
-        float lowerSize = sizeParam.value.get<float>();
-        float higherSize = sizeParam.stops->evalWidth(m_zoom + 1);
```

Comment:
yes - also I think we should use exponential interpolation for all cases that involve zoomlevels